application/modules/book_receive/controllers/Book_receive.php

```diff
@@ -0,0 +1,208 @@
+
+class Book_receive extends Warehouse_Controller
+{
+    public $per_page = 10;
+
+    public function __construct()
+    {
+        parent::__construct();
+        $this->pages = "book_receive";
+        $this->load->model('book_receive/book_receive_model', 'book_receive');
+        // $this->load->model('book_stock/book_stock_model', 'book_stock');
+        // $this->load->model('book_transaction/book_transaction_model', 'book_transaction');
+    }
+
+    //index book receive
+    public function index($page = NULL)
+    {
+        //all filters
+        $filters = [
+            'keyword'           => $this->input->get('keyword', true),
+            'book_receive_status' => $this->input->get('book_receive_status', true)
+        ];
+
+        // custom per page
+        $this->book_receive->per_page = $this->input->get('per_page', true) ?? 10;
+        $get_data = $this->book_receive->filter_book_receive($filters, $page);
+
+        $book_receives = $get_data['book_receives'];
+        $total = $get_data['total'];
+        $pagination = $this->book_receive->make_pagination(site_url('book_receives'), 2, $total);
+
+        $pages      = $this->pages;
+        $main_view  = 'book_receive/index_bookreceive';
+        $this->load->view('template', compact('pages', 'main_view', 'book_receives', 'pagination', 'total'));
+    }
+
+    //edit book receive
+    public function edit($book_receive_id)
+    {
+        $book_receive = $this->book_receive->where('book_receive_id', $book_receive_id)->get();
+        if (!$book_receive) {
+            $this->session->set_flashdata('warning', $this->lang->line('toast_data_not_available'));
+            redirect($this->pages);
+        } else {
+            $pages       = $this->pages;
+            $main_view   = 'book_receive/edit_bookreceive';
+            $this->load->view('template', compact('pages', 'main_view', 'book_receive'));
+        }
+    }
+
+    public function update($book_receive_id)
+    {
+        $book_receive = $this->book_receive->where('book_receive_id', $book_receive_id)->get();
+        $book_receive_id = $this->input->post('book_receive_id');
+        $entry_date = $this->input->post('entry_date');
+        $deadline = $this->input->post('deadline');
+        $finish_date = $this->input->post('finish_date');
+        $is_handover = $this->input->post('is_handover');
+        $handover_start_date = $this->input->post('handover_start_date');
+        $handover_end_date = $this->input->post('handover_end_date');
+        $handover_staff = $this->input->post('handover_staff');
+        $handover_deadline = $this->input->post('handover_deadline');
+        $is_wrapping = $this->input->post('is_wrapping');
+        $wrapping_start_date = $this->input->post('wrapping_start_date');
+        $wrapping_end_date = $this->input->post('wrapping_end_date');
+        $wrapping_staff = $this->input->post('wrapping_staff');
+        $wrapping_deadline = $this->input->post('wrapping_deadline');
```

Comment:
kenapa tidak pakai `$input = (object) $this->input->post(null, false);`? 
ini akan ambil semuanya yang dikirim via form

Reply:
kemarin coba pakai yg `$input = (object) $this->input->post(null, false);` itu malah gak keedit mas terus bingung, jadinya pake yg gini. aku coba2 lagi ya mas pakai `$input = (object) $this->input->post(null, false);`